### PR TITLE
Updates redirects, removes broken links, and updates subsection index entries

### PR DESCRIPTION
**Pull request by discdiver**
### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [ ] If this pull request adds new functionality, it includes unit tests that cover the changes
- [ ] If this pull request removes docs files, it includes redirect settings in `mint.json`.
- [ ] If this pull request adds functions or classes, it includes helpful docstrings.

### CODE REVIEW

File: docs/3.0rc/develop/index.mdx [hunk 1]
```diff
@@ -17,20 +17,17 @@ The **Configure runtime behavior** section introduces advanced features for inte
...
-- [Execute code on state changes](/3.0rc/develop/manage-states#execute-code-on-state-changes) shows how to run code when a task or flow changes state.
 - [Pause and resume flow runs](/3.0rc/develop/pause-resume/) explains how to halt flow runs for input.
 - [Send and receive flow run inputs](/3.0rc/develop/inputs/) highlights how to run Human-in-the-Loop workflows with Prefect.
 - [Access runtime context](/3.0rc/develop/runtime-context/) shows how to access information about the current run.
 
 The **Manage state and configuration** section explores how to manage data and configuration in Prefect.
 
 - [Manage results](/3.0rc/develop/results/) explains how to store and retrieve task results.
-- [Manage states](/3.0rc/develop/manage-states/) discusses how to learn about a workflow through task and flow states.
+- [Manage states](/3.0rc/develop/manage-states/) discusses how to get take action based on task and flow run states.
```
Comment (zzstoatzz) [inline docs/3.0rc/develop/index.mdx:27]:
```suggestion
- [Manage states](/3.0rc/develop/manage-states/) discusses how to take action based on task and flow run states.
```

Comment (cicdw) [inline docs/3.0rc/develop/index.mdx]:
```suggestion
- [Manage states](/3.0rc/develop/manage-states/) discusses how to take action based on task and flow run states.
```

**Review by zzstoatzz — CHANGES_REQUESTED** [review]

File: docs/mint.json [hunk 2]
```diff
@@ -910,7 +910,7 @@
       "source": "/latest/community"
     },
     {
-      "destination": "/latest/develop/interact-with-data",
+      "destination": "/latest/develop",
```
Comment (cicdw) [inline docs/mint.json:913]: this should get covered in `develop/blocks` which I'll revisit today or tomorrow if you want to proactively link there

**Review by cicdw — COMMENTED** [review]
small suggestions

File: docs/mint.json (hunk 2 above)
Comment (discdiver) [reply docs/mint.json:913]: Cool. Done in latest commit.

Comment (cicdw) [reply docs/mint.json:913]: @discdiver i don't think you pushed it yet

Comment (discdiver) [reply docs/mint.json]: My bad, should be there now.

**Review by cicdw — APPROVED** [review]

**Pull request merged**